exchange/bitswap/bitswap_test.go

```diff
@@ -489,3 +490,109 @@ func TestWantlistCleanup(t *testing.T) {
 		t.Fatal("should only have keys[0] in wantlist")
 	}
 }
+
+func assertLedgerMatch(ra, rb *decision.Receipt) error {
+	if ra.Sent != rb.Recv {
+		return fmt.Errorf("mismatch in ledgers (exchanged bytes): %d sent vs %d recvd", ra.Sent, rb.Recv)
+	}
+
+	if ra.Recv != rb.Sent {
+		return fmt.Errorf("mismatch in ledgers (exchanged bytes): %d recvd vs %d sent", ra.Recv, rb.Sent)
+	}
+
+	if ra.Exchanged != rb.Exchanged {
+		return fmt.Errorf("mismatch in ledgers (exchanged blocks): %d vs %d ", ra.Exchanged, rb.Exchanged)
+	}
+
+	return nil
+}
+
+func TestBitswapBytesSentOneWay(t *testing.T) {
+	net := tn.VirtualNetwork(mockrouting.NewServer(), delay.Fixed(kNetworkDelay))
+	sg := NewTestSessionGenerator(net)
+	defer sg.Close()
+	bg := blocksutil.NewBlockGenerator()
+
+	t.Log("Test ledgers match when one peer sends block to another")
+
+	instances := sg.Instances(2)
+	blocks := bg.Blocks(1)
+	err := instances[0].Exchange.HasBlock(blocks[0])
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+	defer cancel()
+	blk, err := instances[1].Exchange.GetBlock(ctx, blocks[0].Cid())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ra := instances[0].Exchange.LedgerForPeer(instances[1].Peer)
+	rb := instances[1].Exchange.LedgerForPeer(instances[0].Peer)
+
+	err = assertLedgerMatch(ra, rb)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	t.Log(blk)
+	for _, inst := range instances {
+		err := inst.Exchange.Close()
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+}
+
+func TestBitswapBytesSentTwoWay(t *testing.T) {
+	net := tn.VirtualNetwork(mockrouting.NewServer(), delay.Fixed(kNetworkDelay))
+	sg := NewTestSessionGenerator(net)
+	defer sg.Close()
+	bg := blocksutil.NewBlockGenerator()
+
+	t.Log("Test ledgers match when two peers send one block to each other")
+
+	instances := sg.Instances(2)
+	blocks := bg.Blocks(2)
+	err := instances[0].Exchange.HasBlock(blocks[0])
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = instances[1].Exchange.HasBlock(blocks[1])
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+	defer cancel()
+	blk, err := instances[1].Exchange.GetBlock(ctx, blocks[0].Cid())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ctx, cancel = context.WithTimeout(context.Background(), time.Second*5)
+	defer cancel()
+	blk, err = instances[0].Exchange.GetBlock(ctx, blocks[1].Cid())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ra := instances[0].Exchange.LedgerForPeer(instances[1].Peer)
```

Comment:
@whyrusleeping makes sense, I saw that in one of the other tests but forget to include it in mine. I add that in the next push, thanks!